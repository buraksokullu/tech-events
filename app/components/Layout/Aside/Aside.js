import React, { Component } from 'react';

import Input from 'Components/FormFields/Input/Input';
import Checkbox from 'Components/FormFields/Checkbox/Checkbox';

import s from './Aside.scss';

export class Aside extends Component {
  changeName = event => {
    const { filterByName } = this.props;

    filterByName(event.target.value);
  };

  filterFreeEvents = event => {
    const { filterFreeEvents } = this.props;

    filterFreeEvents(event.target.checked);
  };

  render() {
    return (
      <div className={s.sidebarHolder}>
        <div className={s.title}>Find your event</div>
        <div className={s.searchInput}>
          <Input
            id="eventName"
            name="eventName"
            placeholder="Name"
            maxLength="50"
            onChange={this.changeName}
          />
        </div>

        <div className={s.chckFilter}>
          <Checkbox
            id="chckOnlyFree"
            name="chckOnlyFree"
            checked={false}
            onChange={this.filterFreeEvents}
            label={(
              <div>
                Only
                <span className={s.isFree}>Free</span>
              </div>
            )}
          />
        </div>
      </div>
    );
  }
}

export default Aside;
