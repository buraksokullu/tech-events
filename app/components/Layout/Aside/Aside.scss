.sidebarHolder {
  display: flex;
  flex-direction: column;
  width: 224px;
  margin-right: 20px;

  @media only screen and (max-width: 1024px) {
    padding-bottom: 10px;
    border-bottom: 1px solid $grey40;
    margin-bottom: 20px;
    width: 100%;
  }

  .title {
    display: none;
    margin-bottom: 10px;
    @media only screen and (max-width: 1024px) {
      display: block;
    }
  }

  .searchInput {
    margin-bottom: 20px;
  }

  .isFree {
    display: inline-block;
    margin-left: 7px;
    position: relative;
    bottom: 1px;
    padding: 4px 8px;
    font-size: 12px;
    color: white;
    background: $greenlight60;
    border-radius: 10px;
  }
}
