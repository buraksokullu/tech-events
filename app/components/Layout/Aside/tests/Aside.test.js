import React from 'react';
import { shallow } from 'enzyme';
import Aside from '../Aside';

describe('Aside', () => {
  let asideProps = null;
  const filterByNameMock = jest.fn();
  const filterFreeEventsMock = jest.fn();
  beforeEach(() => {
    asideProps = {
      filterByName: filterByNameMock,
      filterFreeEvents: filterFreeEventsMock
    };
  });

  it('Should render correctly', () => {
    const component = shallow(<Aside {...asideProps} />);

    expect(component).toMatchSnapshot();
  });

  it('changeName should invoke filterByName with value', () => {
    const event = {
      target: {
        value: 'value'
      }
    };

    const component = shallow(<Aside {...asideProps} />);
    const instance = component.instance();

    instance.changeName(event);
    expect(filterByNameMock).toHaveBeenCalledWith('value');
  });
  it('filterFreeEvents should invoke props filterFreeEvents with value', () => {
    const event = {
      target: {
        checked: 'checked'
      }
    };

    const component = shallow(<Aside {...asideProps} />);
    const instance = component.instance();

    instance.filterFreeEvents(event);
    expect(filterFreeEventsMock).toHaveBeenCalledWith('checked');
  });
});
