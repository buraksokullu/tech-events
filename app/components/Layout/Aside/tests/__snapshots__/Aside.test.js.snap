// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside Should render correctly 1`] = `
<div>
  <div>
    Find your event
  </div>
  <div>
    <Input
      id="eventName"
      input={null}
      maxLength="50"
      name="eventName"
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
  </div>
  <div>
    <CheckBox
      checked={false}
      id="chckOnlyFree"
      input={null}
      label={
        <div>
          Only
          <span>
            Free
          </span>
        </div>
      }
      name="chckOnlyFree"
      onChange={[Function]}
      value={null}
    />
  </div>
</div>
`;
